packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPage.js

```diff
@@ -0,0 +1,388 @@
+      modalDescription,
+      modalSecondaryButtonText,
+      modalTitle,
+      nextButtonText,
+      onClose,
+      onRequestSubmit,
+      submitButtonText,
+      toggleAriaLabel,
+      toggleLabelText,
+      ...rest
+    },
+    ref
+  ) => {
+    const [createFullPageActions, setCreateFullPageActions] = useState([]);
+    const [currentStep, setCurrentStep] = useState(1);
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [open, setOpen] = useState(false);
+
+    useEffect(() => {
+      const onUnmount = () => {
+        setCurrentStep(0);
+        setIsSubmitting(false);
+        onClose();
+      };
+      const handleOnRequestSubmit = async () => {
+        // check if onRequestSubmit returns a promise
+        try {
+          await onRequestSubmit();
+          onUnmount();
+        } catch (error) {
+          setIsSubmitting(false);
+          console.warn(`${componentName} submit error: ${error}`);
+        }
+      };
+      const isSubmitDisabled = () => {
+        let step = 0;
+        let submitDisabled = false;
+        const createFullPageSteps = getFullPageSteps();
+        createFullPageSteps.forEach((child) => {
+          step++;
+          if (currentStep === step) {
+            submitDisabled = child.props.disableSubmit;
+          }
+        });
+        return submitDisabled;
+      };
+      const handleNext = async () => {
+        setIsSubmitting(true);
+        const createSteps = getFullPageSteps();
+        if (createSteps[currentStep - 1].props.onNext) {
+          try {
+            await createSteps[currentStep - 1].props.onNext();
+            continueToNextStep();
+          } catch (error) {
+            setIsSubmitting(false);
+            console.warn(`${componentName} onNext error: ${error}`);
+          }
+        } else {
+          continueToNextStep();
+        }
+      };
+      const handleSubmit = async () => {
+        setIsSubmitting(true);
+        const createSteps = getFullPageSteps();
+        // last step should have onNext as well
+        if (createSteps[currentStep - 1].props.onNext) {
+          try {
+            await createSteps[currentStep - 1].props.onNext();
+            await handleOnRequestSubmit();
+          } catch (error) {
+            setIsSubmitting(false);
+            console.warn(`${componentName} onNext error: ${error}`);
+          }
+        } else {
+          await handleOnRequestSubmit();
+        }
+      };
+      if (getFullPageSteps()?.length) {
+        const createSteps = getFullPageSteps();
+        const total = createSteps.length;
+        const buttons = [];
+        if (total > 1) {
+          buttons.push({
+            label: backButtonText,
+            onClick: () => setCurrentStep((prev) => prev - 1),
+            kind: 'secondary',
+            disabled: currentStep === 1,
+          });
+          buttons.push({
+            label: cancelButtonText,
+            onClick: () => {
+              setOpen(true);
```

Comment:
The `open` value should be set in the component that is using `<CreateFullPage />` most likely. In CreateTearsheet, you'll see that the `Cancel` button calls the `onClose` prop.

Reply:
This `open` refers to the "Confirmation modal" that is triggered when you press Cancel. The component itself will not be triggered by a button or an open -- it is a layout.

Reply:
Ok, I understand that this should trigger the modal to show. See my comment [here](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/888/files#r649993453).